Fix #27056: wrong layout ordering when excerpt exists

prodoelmit opened the pull request:
Resolves: #27056

`LayoutPanelTreeModel` receives masterParts that are already sorted as they are in score. So there is no need to try to reorder them based on some other criteria. 

(Though I'm not completely sure about it, so please let me know if I've missed something)

- [x] I signed the [CLA](https://musescore.org/en/cla)
- [x] The title of the PR describes the problem it addresses
- [x] Each commit's message describes its purpose and effects, and references the issue it resolves
- [ ] If changes are extensive, there is a sequence of easily reviewable commits
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] There are no unnecessary changes
- [x] The code compiles and runs on my machine, preferably after each commit individually
- [ ] I created a unit test or vtest to verify the changes I made (if applicable)

prodoelmit commented on src/instrumentsscene/view/layoutpaneltreemodel.cpp:
```diff
@@ -401,7 +401,8 @@ void LayoutPanelTreeModel::load()
     async::NotifyList<const Part*> masterParts = m_masterNotation->parts()->partList();
-    sortParts(masterParts);
+    // We get parts from score where it's already well-ordered
+    //sortParts(masterParts);
```
@RomanPudashkin thanks for explanation! I'll check how I can fix it better

prodoelmit replied:
Pushed two commits. 

First one (58d6542) simply ignores reordering only for master score. 
Second (7b24d00) uses order of parts in excerpt's score to sort partList from master score. This way we get full list of parts but in locally correct order - all without storing any additional storing mechanisms (custom ordering list can be safely removed if this way is ok)

DmitryArefiev commented:
Tested #27056 on Win10, Mac13.7.8 - **FIXED**

RomanPudashkin commented on src/instrumentsscene/view/layoutpaneltreemodel.cpp:
```diff
@@ -442,6 +451,49 @@ void LayoutPanelTreeModel::load()
+    }
+
+    LOGI("-----");
+    LOGI("Reference order:");
```
@prodoelmit could you please remove all these logs?

prodoelmit replied:
Yeah sure. If second way is ok for you, I'll prepare cleaned version

prodoelmit commented:
@RomanPudashkin, cleaned it. 

While cleaning, I've also noticed couple bad decisions:

* Using usual `sort` may cause resorting of parts not enabled in part's view. Switched to `stable_sort` instead
* Comparing raw iterators isn't guaranteed by standard, so I've switched to comparing `std::distance`
